cloud/interfaces.go

```diff
@@ -71,3 +66,17 @@ type ClusterDescriber interface {
 	NodeRouteTable() *infrav1.RouteTable
 	ControlPlaneRouteTable() *infrav1.RouteTable
 }
+
+// ClusterDescriber is an interface which can get common Azure Cluster information.
+type ClusterDescriber interface {
+	ResourceGroup() string
+	ClusterName() string
+	Location() string
+	AdditionalTags() infrav1.Tags
+}
+
+type ClusterScope interface {
```

Comment:
Feels a little odd for `ClusterScope` to be an interface when all the other `*Scope` types are structs.

Reply:
Indeed... naming is hard

I need one thing that describes the whole k8s Cluster with networking, and one that describes the basic cluster info like resource group and cluster name. Any suggestions?

Reply:
`ClusterScoper`? 🤔

Reply:
sgtm

Reply:
`ClusterPlan`?

Reply:
I think I like ClusterScoper better than ClusterPlan because it describes what it is more closely and follows the interface naming convention by ending with `er`